#include "EditSource.h"
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
/// @name  //
/// @brief //
/// @param //
//
EditSource::EditSource(QWidget *parent)
    : QWidget(parent)
{
  this->setupUi(this);
  // update edit fonts:
  QFont fontAtoms(editAtoms_->font());
  this->updateFont(fontAtoms);
}
void EditSource::updateFont(QFont f)
{
  editAtoms_->setFont(f);
  editTitle_->setFont(f);
  editFormula_->setFont(f);
  editNumberOfAtoms_->setFont(f);
}
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
/// @name  //
/// @brief //
/// @param //
//
bool EditSource::isModified(void) const
{
  return editTitle_->isModified() || editAtoms_->isModified();
}
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
/// @name  //
/// @brief //
/// @param //
//
QString EditSource::getTitleString() const
{
  return editTitle_->text();
}
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
/// @name  //
/// @brief //
/// @param //
//
void EditSource::resetTitleString(QString str)
{
  editTitle_->setText(str);
}
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
/// @name  //
/// @brief //
/// @param //
//
EditMarkLines *EditSource::getEditAtoms() const
{
  return editAtoms_;
}
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
/// @name  //
/// @brief //
/// @param //
//
Molecule *EditSource::resetMolecule(Molecule *pMol)
{
    if(pMol != ptrMolecule_)
        std::swap(ptrMolecule_,pMol);
    return pMol;
}
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
/// @name  //
/// @brief //
/// @param //
//
void EditSource::showMolecule()
{

}
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////
